socs/agents/ucsc_radiometer/agent.py

```diff
@@ -0,0 +1,142 @@
+                                 buffer_time=1
+                                 )
+
+        self.last_published_reading = None
+
+    @ocs_agent.param('test_mode', default=False, type=bool)
+    def acq(self, session, params=None):
+        """acq()
+
+        **Process** - Fetch values from PWV Flask Server
+
+        Parameters
+        ----------
+        test_mode : bool, option
+            Run the Process loop only once. Meant only for testing.
+            Default is False.
+
+        Notes
+        -----
+        The most recent data collected is stored in session data in the
+        following structure::
+
+            >>> response.session['data']
+            {'timestamp': 1678820419.0,
+             'pwv': 0.49253026985972237}
+
+        """
+        pm = Pacemaker(1 / 60, quantize=False)
+
+        self.take_data = True
+        while self.take_data:
+            r = requests.get(self.url)
+            data = r.json()
+            last_pwv = data['pwv']
+            last_timestamp = data['timestamp']
+
+            pwvs = {'block_name': 'pwvs',
+                    'timestamp': last_timestamp,
+                    'data': {'pwv': last_pwv}
+                    }
+
+            if self.last_published_reading is not None:
+                if last_timestamp > self.last_published_reading[1]:
+                    self.agent.publish_to_feed('pwvs', pwvs)
+                    self.last_published_reading = (last_pwv, last_timestamp)
+            else:
+                self.agent.publish_to_feed('pwvs', pwvs)
+                self.last_published_reading = (last_pwv, last_timestamp)
+
+            session.data = {"timestamp": last_timestamp,
+                            "pwv": {}}
+
+            session.data['pwv'] = last_pwv
+
+            if params['test_mode']:
+                break
+            else:
+                pm.sleep()
+
+        return True, 'Acquisition exited cleanly.'
+
+    def _stop_acq(self, session, params=None):
+        """
+        Stops acq process.
+        """
+        self.take_data = False
+        return True, 'Stopping acq process'
+
+
+def add_agent_args(parser=None):
+    if parser is None:
+        parser = argparse.ArgumentParser()
+    pgroup = parser.add_argument_group('Agent Options')
+    pgroup.add_argument("--url", type=str, help="url for radiometer web server")
+    pgroup.add_argument("--test_mode", type=bool, help="True for setting agent in test_mode. Default is False.")
+    return parser
+
+
+def main(args=None):
+    # For logging
+    txaio.use_twisted()
+    txaio.make_logger()
+
+    txaio.start_logging(level=environ.get("LOGLEVEL", "info"))
+
+    parser = add_agent_args()
+    args = site_config.parse_args(agent_class='UCSCRadiometerAgent', parser=parser, args=args)
+
+    # test params
+    test_params = False
+    if args.test_mode:
+        test_params = {'acq_params': {'test_mode': True}}
+    elif not args.test_mode:
+        test_params = {'acq_params': {'test_mode': False}}
```

Comment:
This should be revisited. The `dict` passed to `startup` in `agent.register_process()` should contain the keyword arguments for that process. `acq_params` isn't an argument for `acq` in this agent, so what's happening during your test run is `acq` tries to startup on agent launch, doesn't run because of an unexpected parameter, and then your test starts `acq` with the `test_mode` argument and things look like they work. Here's a snippet from the agent logs when running the test:
```
2023-08-29T23-24-59.825134 startup-op: launching acq
2023-08-29T23-24-59.825323 start called for acq
2023-08-29T23-24-59.825503 Caught ParamError during start call: params included unexpected values: ['acq_params']
2023-08-29T23-25-00.066828 start called for acq
2023-08-29T23-25-00.067338 acq:0 Status is now "starting".
2023-08-29T23-25-00.074380 acq:0 Acquisition exited cleanly.
2023-08-29T23-25-00.074763 acq:0 Status is now "done".
```